instrumentation/spring/spring-webmvc/spring-webmvc-6.0/library/build.gradle.kts

```diff
@@ -15,3 +15,16 @@ dependencies {
 otelJava {
   minJavaVersionSupported.set(JavaVersion.VERSION_17)
 }
+
+tasks {
+  compileJava {
+    // We compile this module for java 8 because it is used as a dependency in spring-boot-autoconfigure.
+    // If this module is compiled for java 17 then gradle can figure out based on the metadata that
+    // spring-boot-autoconfigure has a dependency that requires 17 and fails the build when it is used
+    // in a project that targets an earlier java version.
```

Comment:
very interesting, gradle being too smart for us...

cc @jeanbisutti @zeitlinger